src/vcs/git.rs

```diff
@@ -192,6 +204,65 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
```

Comment:
Change to `expect()`?

Reply:
I think we can `?`. I'm pretty sure we have file path errors in the top-level `Error` type. I'd like to remove `expect` above as well :)